middleware/context.go

```diff
@@ -86,6 +87,16 @@ func IsSkipRateLimit(ctx context.Context) bool {
 	return ok
 }
 
+// WithOverridePublicLimit override rate limiting configuration.
+func WithOverridePublicLimit(ctx context.Context, limit int) context.Context {
```

Comment:
can you rename to `OverridePublicRateLimit` and the Get method too..

cuz its not clear what "limit" this is..